app/services/hashtag_query_service.rb

```diff
@@ -2,15 +2,14 @@
 
 class HashtagQueryService < BaseService
   def call(tag, params, account = nil, local = false)
-    any  = tags_for(params[:any])
+    tags = tags_for(Array(tag.name) | Array(params[:any])).pluck(:id)
     all  = tags_for(params[:all])
     none = tags_for(params[:none])
 
-    @query = Status.as_tag_timeline(tag, account, local)
-                   .tagged_with_all(all)
-                   .tagged_with_none(none)
-    @query = @query.distinct.or(self.class.new.call(any, params.except(:any), account, local).distinct) if any
-    @query
+    Status.distinct
+          .as_tag_timeline(tags, account, local)
```

Comment:
What happens if `tags` is nil when passed here?

Reply:
Hmm, well I was assuming here that 'tag' would always be passed into the method, so
```
tags = tags_for(Array(tag.name))
```
_should_ always be present.

If it did end up being empty (if nil were passed in as the `tag` param, for example), I believe tags would go into `Status#tagged_with`,
```
  scope :tagged_with, ->(tag) { joins(:statuses_tags).where(statuses_tags: { tag_id: tag }) }
```
which should return an empty relation